apps/openmw/mwmechanics/spellcasting.cpp

```diff
@@ -1064,21 +1086,34 @@ namespace MWMechanics
             adjustDynamicStat(creatureStats, effectKey.mId-ESM::MagicEffect::RestoreHealth, magnitude);
             break;
         case ESM::MagicEffect::DamageHealth:
-            receivedMagicDamage = true;
-            adjustDynamicStat(creatureStats, effectKey.mId-ESM::MagicEffect::DamageHealth, -magnitude);
-            break;
+            if (!godmode)
+            {
+                receivedMagicDamage = true;
+                adjustDynamicStat(creatureStats, effectKey.mId-ESM::MagicEffect::DamageHealth, -magnitude);
+            }
         case ESM::MagicEffect::DamageMagicka:
         case ESM::MagicEffect::DamageFatigue:
-            adjustDynamicStat(creatureStats, effectKey.mId-ESM::MagicEffect::DamageHealth, -magnitude);
+            if (!godmode)
+            {
+                adjustDynamicStat(creatureStats, effectKey.mId-ESM::MagicEffect::DamageHealth, -magnitude);
+            }
+
             break;
+
         case ESM::MagicEffect::AbsorbHealth:
-            if (magnitude > 0.f)
-                receivedMagicDamage = true;
-            adjustDynamicStat(creatureStats, effectKey.mId-ESM::MagicEffect::AbsorbHealth, -magnitude);
-            break;
```

Comment:
Is this break missing or there is some reason to remove it? Found by warning from gcc 7. Same at line [1069](https://github.com/OpenMW/openmw/pull/1245/commits/14b59e0e4be30b54e2a84493a34c07140ddf92fc#diff-87f3095511c5b627b0da19a137d5a7a8L1069).
```
/home/elsid/dev/openmw/apps/openmw/mwmechanics/spellcasting.cpp: In function ‘bool MWMechanics::effectTick(MWMechanics::CreatureStats&, const MWWorld::Ptr&, const MWMechanics::EffectKey&, float)’:
/home/elsid/dev/openmw/apps/openmw/mwmechanics/spellcasting.cpp:1089:13: warning: this statement may fall through [-Wimplicit-fallthrough=]
             if (!godmode)
             ^~
/home/elsid/dev/openmw/apps/openmw/mwmechanics/spellcasting.cpp:1094:9: note: here
         case ESM::MagicEffect::DamageMagicka:
         ^~~~
/home/elsid/dev/openmw/apps/openmw/mwmechanics/spellcasting.cpp:1104:13: warning: this statement may fall through [-Wimplicit-fallthrough=]
             if (!godmode)
             ^~
/home/elsid/dev/openmw/apps/openmw/mwmechanics/spellcasting.cpp:1110:9: note: here
         case ESM::MagicEffect::AbsorbMagicka:
         ^~~~
```